src/acceptance/assets/app/go_app/deploy.sh

```diff
@@ -1,37 +1,53 @@
 #!/bin/bash
-
 set -euo pipefail
-config=${CONFIG:-}
+
+# read content of config right at the beginning to avoid errors after switching directories
+config="$(cat "${CONFIG:-}" 2> /dev/null || echo "")"
+
 function getConfItem(){
-  val=$(jq -r ".$1" "${config}")
+  val=$(jq -r ".$1" <<< "$config")
   if [ "$val" = "null" ]; then return 1; fi
   echo "$val"
 }
-if [ -z "${config}" ]; then
+
+if [ -z "$config" ]; then
   echo "ERROR: Please supply the config using CONFIG env variable"
   exit 1
 fi
 
-script_dir=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
-app_dir="$( realpath -e "${script_dir}/build")"
-service_name=$(getConfItem service_name)
-
-cf create-org test
-cf target -o test
-cf create-space test_app
-cf target -s test_app
-
-cf enable-service-access "${service_name}" -b "$(getConfItem service_broker)" -p  "$(getConfItem service_plan)" -o test
-cf create-service "${service_name}" "$(getConfItem service_plan)" "${service_name}" -b "$(getConfItem service_broker)" --wait
-
-pushd "${app_dir}" >/dev/null
-cf push\
-  --var app_name=test_app\
-  --var app_domain="$(getConfItem apps_domain)"\
-  --var service_name="${service_name}"\
-  --var instances=1\
-  --var memory_mb="$(getConfItem node_memory_limit||echo 128)"\
-  --var buildpack="binary_buildpack"\
-  -f "manifest.yml"\
-  -c ./app
-popd > /dev/null
\ No newline at end of file
+org="test"
+space="test_$(whoami)"
+cf create-org "$org"
+cf target -o "$org"
+cf create-space "$space"
+cf target -s "$space"
+
+app_name="test_app"
+app_domain="$(getConfItem apps_domain)"
+service_name="$(getConfItem service_name)"
+memory_mb="$(getConfItem node_memory_limit||echo 128)"
+service_broker="$(getConfItem service_broker)"
+service_plan="$(getConfItem service_plan)"
+
+cf enable-service-access "$service_name" -b "$service_broker" -p  "$service_plan" -o test
+cf create-service "$service_name" "$service_plan" "$service_name" -b "$service_broker" -t "app-autoscaler" --wait
```

Comment:
Provide a tag so that the app is able to find the service, see also https://github.com/cloudfoundry/app-autoscaler-release/blob/781fba30707e779a46279b4e008b7f559587fe83/src/acceptance/assets/app/go_app/internal/app/custom_metrics.go#L122

Reply:
Interesting, was this an issue? It should have been set via the broker: https://github.com/cloudfoundry/app-autoscaler-release/blob/main/templates/app-autoscaler.yml#L293

Reply:
Very good point! You are right, the problem was that the app wasn't restaged / created upfront.